File: go-patterns/creational/factory/tmpl/Operator.go
package tmpl

// Operator Product
type Operator interface {
	calc(n1, n2 int) int
}

// Add Concrete Product
type Add struct {
}

func (a Add) calc(n1, n2 int) int {
	return n1 + n2
}

// Mul Concrete Product
type Mul struct {
}

func (a Mul) calc(n1, n2 int) int {
	return n1 * n2
}

File: java-patterns/src/main/java/creational/factory/tmpl/Operator.java
package creational.factory.tmpl;

/**
 * Usage: Product <br/>
 * Date: 2023/4/11 18:00 <br/>
 *
 * @author <a href="mailto:hanzhang2566@foxmail.com">hanzhang</a>
 */
public interface Operator {
    /**
     * 用来计算的方法
     *
     * @param n1 number1
     * @param n2 number2
     * @return result
     */
    int calc(int n1, int n2);
}
